src/functions/edi/inbound/handler.ts

```diff
@@ -102,34 +102,50 @@ export const handler = async (event: any): Promise<Record<string, any>> => {
             receivingPartnerId,
           });
 
-          for (const functionalGroup of interchange.functionalGroups) {
-            const guideIdsForPartnership = transactionSetConfigs.map(
-              (config) => config.guideId
-            );
+          const guideIdsForPartnership = transactionSetConfigs.map(
+            (config) => config.guideId
+          );
 
+          for (const functionalGroup of interchange.functionalGroups) {
             // For each Transaction Set:
-            // - look up the guideId and mappingId
-            // - call processTransactionSet, which translates X12 to JSON, and then invokes the mapping
-            // - send the result to the webhook
-            // - delete the input file once processed successfully
-
+            // - look up the guideId
+            // - find the corresponding transaction set config from the partnership
+            // - call processEdi, which translates X12 to JSON
+            // - call deliverToDestination for each destination, which does the following:
+            //   - optionally invokes the mapping if one is included in config
+            //   - sends the result to the destination
             for (const transactionSet of functionalGroup.transactionSets) {
               // load the guide for the transaction set
               const guideSummary = await resolveGuide({
                 guideIdsForPartnership,
                 transactionSetType: transactionSet.transactionSetType,
               });
 
-              const edi = transactionSet.segments.join(
+              const transactionSetContents = transactionSet.segments.join(
                 interchange.delimiters.segment
               );
 
+              const documentContents = [
```

Comment:
hydrate full edi document for transaction set by including interchange and functional group segments

Reply:
adding a question that @tommy raised here as well:

> So, you’re translating each ST separately, and duplicating envelope to wrap it before sending it to Translate? Just wondering, is there any issue with repeating the control numbers in that way?

my response was:
> we're actually translating each GS separately, but it's a good question... I think it's likely ok, but let me mention it in the PR for @BDQ to weigh in. for the most part I think customers just want to translate the EDI to JSON, and then pass that data to whatever destination.
> 
> if we're going to look at control numbers for duplicate detection, etc., I think we'd do that at the interchange/functional group level of the logic (before we actually call translate & start delivering output to destinations)

lmk what your thoughts are, Brian!

Reply:
thinking about this further, beyond the control numbers question that Tommy raised, it would probably be best to not change the structure of the input when we send the data to the destination.

IMO it would be best for each interchange to be processed and sent to the destinations in its entirety. will take a stab at this refactor and update the PR

Reply:
I don't think there's any harm restructuring the input, and duplicating the envelope - we do this with RT, the envelope info is generally only needed for referencing some sender or time information.

I certainly do not think we should process the entire Interchange, in reality that's way too much potential to have a single txn that failure on one destination in a large batch.

Reply:
Closing the loop on this conversation: I discussed with Brian and will be leaving the current logic in place (each transaction will be sent to destinations separately, with reused envelope data as needed)